.github/workflows/ci.yml

```diff
@@ -144,10 +144,95 @@ jobs:
+
+      - name: Docker Buildx
+        uses: docker/setup-buildx-action@ecf95283f03858871ff00b787d79c419715afc34 # v2.7.0
+      
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@818d4b7b91585d195f67373fd9cb0332e31a7175 # v4.6.0
+        with:
+          images: |
+            name=ghcr.io/nginxinc/nginx-kubernetes-gateway
+          tags: |
+            type=semver,pattern={{version}}
+            type=edge
+            type=ref,event=pr
+            type=ref,event=branch,suffix=-rc,enable=${{ startsWith(github.ref, 'refs/heads/release') }}
+
+      - name: Prepare NKG files
+        run: |
+          nkg_prefix=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 1)
+          nkg_tag=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 2)
+          yq -i "with(.spec.template.spec.containers[0]; .image = \"${nkg_prefix}:${nkg_tag}\" | .imagePullPolicy = \"Never\")" deploy/manifests/deployment.yaml
+        shell: bash
+
+      - name: Build Docker Image
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        with:
+          file: build/Dockerfile
+          tags: ${{ steps.meta.outputs.tags }}
+          context: "."
+          target: goreleaser
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          pull: true
+      
+      - name: Build Test Docker Image
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        with:
+          file: conformance/tests/Dockerfile
+          tags: conformance-test-runner:${{ github.sha }}
+          context: "."
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          pull: true
+
+      - name: Deploy Kubernetes
+        id: k8s
+        run: |
+          kind create cluster --image kindest/node:v1.27.1 --kubeconfig kube-${{ github.run_id }}
+          echo "KUBECONFIG=kube-${{ github.run_id }}" >> "$GITHUB_ENV"
+        working-directory: ./conformance
+        shell: bash
+
+      - name: Setup conformance tests
+        run: |
+          nkg_prefix=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 1)
+          nkg_tag=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 2)
+          kind load docker-image ${nkg_prefix}:${nkg_tag}
+          make preload-nginx-container
+          make prepare-nkg-dependencies
+          yq '(select(di != 3))' provisioner/provisioner.yaml | kubectl apply -f -
+          yq "(select(.spec.template.spec.containers[].image) | .spec.template.spec.containers[].image=\"${nkg_prefix}:${nkg_tag}\" | .spec.template.spec.containers[].imagePullPolicy = \"Never\")" provisioner/provisioner.yaml | kubectl apply -f -          
+        working-directory: ./conformance
+        shell: bash
+
+      - name: Run conformance tests
+        run: |
+          make run-conformance-tests TAG=${{ github.sha }}
```

Comment:
We might want to add `continue-on-error: true` here until we fix the failing tests.

Reply:
Is there a way to only run specific tests? That way we can have a passing pipeline? Otherwise, we'll have to allow it to fail so we can still merge things.

If enabling specific tests is an option though, we'll have to enable more tests every time we complete a feature that causes a new test to pass.

Reply:
one way to do that is https://github.com/nginxinc/nginx-kubernetes-gateway/blob/fa56b48670c98283ddc2801d81041401ac1ecaa5/tests/conformance_test.go#L42-L47 (in those lines, only that specific test is run)

Reply:
Maybe setting "continue-on-error" is fine for now - I'm not sure if figuring out skipping individual tests is worth the effort at this stage when we only have a few things to fix, and it feels out of scope for this PR I think. We can remove "continue-on-error", and add passing conformance tests to the required steps, when the current suite is passing. I can create an issue to follow up on those as well.